ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/StateRef.kt

```diff
@@ -0,0 +1,13 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.crypto.SecureHash
+
+/**
+ * Defines a reference to a [ContractState].
+ *
+ * @property index The index of the state in the transaction's outputs in which the referenced state was created.
+ * @property transactionHash The hash of the transaction in which the referenced state was created.
+ */
+@CordaSerializable
+data class StateRef(val transactionHash: SecureHash, val index: Int)
```

Comment:
We need a clear toString that can reversibly be parsed back, because people are doing this quite a lot. I would suggest delegating to JsonMarshallingService, but not sure if notary would handle that.